lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1097,6 +1469,272 @@ void HdVP2Material::_ApplyVP2Fixes(HdMaterialNetwork& outNet, const HdMaterialNe
     }
 }
 
+#ifdef WANT_MATERIALX_BUILD
+
+void HdVP2Material::_ApplyMtlxVP2Fixes(HdMaterialNetwork2& outNet, const HdMaterialNetwork2& inNet)
```

Comment:
Can we share any of the code in _ApplyMtlxVP2Fixes and _ApplyVP2Fixes? Maybe we could make some helper functions that replace the names or something? If we have to make any changes to that system for re-using shaders we'll have to remember to update both code paths and they are far enough apart now that it seems like we might forget.

Reply:
The main issue is that the UsdPreviewSurface code is using the old HdMaterialNetwork, while the MaterialX version uses the more recent HdMaterialNetwork2.
These two functions will have a lot of shareable code once we update everything to the latest material API.